packages/components/src/utils/use-update-effect.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useEffect } from '@wordpress/element';
+
+/*
+ * A `React.useEffect` that will not run on the first render.
+ * Source:
+ * https://github.com/reakit/reakit/blob/master/packages/reakit-utils/src/useUpdateEffect.ts
+ */
+export function useUpdateEffect( effect, deps ) {
```

Comment:
Do we use this technique in other parts of our code base? would it make sense to move it to compose package or is it too soon for it to be a public API?

Reply:
I believe I may have used it in my component work. I know I've had to use it many times in other projects. I chatted about it briefly with @diegohaz . It sounds like he shares the same sentiment, which is why it exists as a Reakit util I suppose :).

I'm be open to adding it to `@wordpress/compose`.

Side question... I'm starting to build up a collection of utils within `@wordpress/components`. There are more that I'd like to add/abstract. In the future, would some of these migrate to something like `@wordpress/compose`? Or would we create a new dedicated package for these JS/component utils?

Thanks!

Reply:
I think compose makes sense for React hooks. The question becomes when are we confident that we'll maintain backward compatibility forever. So it seems to me starting as internal to components and moving to compose when we think they're stable makes sense.

Now, for other kind of utils, i'd say it depends, we generally avoid generic packages like "utils" or "helpers" or things like that and prefer more dedicated packages "url", "priority-query"...